### PR TITLE
Make splits happen via raft command

### PR DESCRIPTION
**Pull request by tylerwilliams**
Make splits happen via raft command.

### CODE REVIEW

File: enterprise/server/raft/replica/replica.go [hunk 1]
```diff
@@ -819,6 +824,74 @@ func (sm *Replica) printRange(r pebble.Reader, iterOpts *pebble.IterOptions, tag
 	}
 }
 
+// REMEMBER: no raft commands may runwhile this one is running!
+// xxx Request contains the new ClusterID and RangeID
```
Comment (vadimberezniker) [inline enterprise/server/raft/replica/replica.go]: Clean up the comment.

File: enterprise/server/raft/store/store.go [hunk 2]
```diff
@@ -1062,6 +1058,97 @@ func (s *Store) TransferLeadership(ctx context.Context, req *rfpb.TransferLeader
...
+		default:
+			if rd := s.lookupRange(clusterID); rd != nil {
+				return nil
+			}
+			time.Sleep(10 * time.Millisecond)
+		}
+	}
+}
+
+func (s *Store) SnapshotCluster(ctx context.Context, clusterID uint64) error {
```
Comment (vadimberezniker) [inline enterprise/server/raft/store/store.go:1075]: I'd add a comment to clarify that this a local operation (i.e. it doesn't affect other nodes)

**Review by vadimberezniker — APPROVED** [review]

File: enterprise/server/raft/store/store.go (hunk 2 above)
Comment (tylerwilliams) [reply enterprise/server/raft/store/store.go:1077]: Done.

File: enterprise/server/raft/replica/replica.go (hunk 1 above)
Comment (tylerwilliams) [reply enterprise/server/raft/replica/replica.go]: Done.

**Pull request merged**